examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -3,87 +3,86 @@
 Poisson regression and non-normal loss
 ======================================
 
-This example illustrates the use of log-linear Poisson regression
-on the `French Motor Third-Party Liability Claims dataset
-<https://www.openml.org/d/41214>`_ from [1]_ and compares
-it with models learned with least squared error. In this dataset, each sample
-corresponds to an insurance policy, i.e. a contract within an insurance
-company and an individual (policiholder). Available features include driver
-age, vehicle age, vehicle power, etc.
-
-A few definitions: a *claim* is the request made by a policyholder to the
-insurer to compensate for a loss covered by the insurance. The *exposure* is
-the duration of the insurance coverage of a given policy, in years.
-
-Our goal is to predict the expected number of insurance claims (or frequency)
-following car accidents for a policyholder given the historical data over a
-population of policyholders.
+This example illustrates the use of log-linear Poisson regression on the
+`French Motor Third-Party Liability Claims dataset
+<https://www.openml.org/d/41214>`_ from [1]_ and compares it with models
+learned with least squared error.
+
+A few definitions:
+
+- a **policy** is a contract between an insurance company and an individual:
+  the **policyholder**, that is, the vehicle driver in this case.
+
+- a **claim** is the request made by a policyholder to the insurer to
+  compensate for a loss covered by the insurance.
+
+- the **exposure** is the duration of the insurance coverage of a given policy,
+  in years.
+
+- the claim **frequency** is the number of claims divided by the exposure,
+  typically measured in number of claims per year.
+
+In this dataset, each sample corresponds to an insurance policy. Available
+features include driver age, vehicle age, vehicle power, etc.
+
+Our goal is to predict the expected frequency of claims following car accidents
+for a new policyholder given the historical data over a population of
+policyholders.
 
```

Comment:
Let us please agree on not having that discussion here. I understand and admire your point of view and would be very glad to talk with you about this controversial (and never ending) subject. At last, note that neither gender nor race are in the data. :no_fitting_emoji:

Reply:
The issue is that the data is biased and we can't even measure that bias because we don't even have the features we need (like race and gender). Not using those features doesn't mean we're not going to have biased model discriminating against a certain group, and I'm very very worried about putting an example out there which people would then use as a reference to [unintentionally] discriminate against people. @romanlutz, do you happen to have a good example for this one?

Reply:
I'm all for better examples for controlling for bias, but I also don't think this PR is the right place for this discussion. It merely refactors an existing example, we should have this discussion in a separate issue.

As a side note, I imagine there could indeed be sample selection some bias in the the data (i.e. company chooses customers), however the target variable (frequency or cost of accidents) shouldn't be too biased, I think?  At least significantly less biased than in other examples such as https://scikit-learn.org/dev/auto_examples/inspection/plot_linear_model_coefficient_interpretation.html for predicting wage. Also pricing policy of an insurance company doesn't directly impact how often one has accidents, so at least there is no immediate feedback loop with the training data. I'm not an expert on this, and there are likely other effects, but I'm just saying that we should discuss it in a separate issue.

Reply:
I don't disagree with the suggestions to discuss this elsewhere. For more information on what @adrinjalali is referring to I recommend "Big Data's Disparate Impact" by Barocas and Selbst. file:///C:/Users/rolutz/AppData/Local/Temp/SSRN-id2477899.pdf section I.D. The entire piece is actually relevant for such a scenario, but that's the section that discusses excluding sensitive features such as race & gender. Other than that there's still potential bias in how the data got collected (I.A, I.B, I.C).

I think there's value in acknowledging such potential shortcomings so that people don't assume that it's the best (or only) way to approach the task. We wouldn't want users to end up on my [list of questionable or unethical use cases](https://github.com/romanlutz/ResponsibleAI). With the acknowledgment it should be clear that it's just a demonstration of scikit-learn. At least that's my point of view :-) Thanks for asking!

Reply:
@adrinjalali and @romanlutz I opened #16715 if you want to address this topic further.